.github/workflows/ci.yml

```diff
@@ -247,6 +249,10 @@ jobs:
     timeout-minutes: 5
     needs: build
```

Comment:
Can we make these depend on the specific build job (relay/relay-pop) as opposed to the entire `build` matrix? Currently, it looks like this:

![image](https://github.com/getsentry/relay/assets/1433023/f69c92f0-e7e3-4312-b86f-3833639f0458)

This would be especially useful for the "Sentry-Relay Integration Tests", which only need to depend on the "relay" build step, but not on "relay-pop".

Reply:
On some quick research, it seems "[Reusable Workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-a-matrix-strategy-with-a-reusable-workflow)" would be the solution. That's too big of a change for now, so I'm fine if we stick with this.